packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -0,0 +1,648 @@
+                    disabled={saving || !(hasName && hasRequiredValues && !loadingData)}
+                >
+                    {saving
+                        ? savedChartModel
+                            ? 'Saving Chart...'
+                            : 'Creating Chart...'
+                        : savedChartModel
+                          ? 'Save Chart'
+                          : 'Create Chart'}
+                </button>
+            </FormButtons>
+        );
+    }
+
+    return (
+        <Modal
+            className="chart-builder-modal"
+            onCancel={onHide_}
+            title={savedChartModel ? 'Edit Chart' : 'Create Chart'}
+            footer={footer}
+        >
+            {error && <Alert>{error}</Alert>}
+            <div className="row">
+                <div className="col-xs-1 col-left">
+                    {chartTypes.map(type => {
+                        const selected = selectedType.name === type.name;
+                        const selectable = !savedChartModel && selectedType.name !== type.name;
+
+                        return (
+                            <div
+                                key={type.name}
+                                className={classNames('chart-builder-type', { selected, selectable })}
+                                data-name={type.name}
+                                onClick={onChartTypeChange}
+                            >
+                                <SVGIcon
+                                    height={null}
+                                    iconSrc={!selected ? ICONS[type.name] + '_gray' : ICONS[type.name]}
+                                />
+                                <div className="title">{type.title}</div>
+                            </div>
+                        );
+                    })}
+                </div>
+                <div className="col-xs-11 col-right">
+                    <div className="chart-type-inputs">
+                        <div className="row">
+                            <div className="col-xs-4">
+                                <label>Name *</label>
+                                <input
+                                    className="form-control"
+                                    name="name"
+                                    placeholder="Enter a name"
+                                    type="text"
+                                    onChange={onNameChange}
+                                    value={name}
+                                />
+                                {canShare && (
+                                    <div className="checkbox-input">
+                                        <input
+                                            name="shared"
+                                            type="checkbox"
+                                            checked={shared}
+                                            onChange={onToggleShared}
+                                        />{' '}
```

Comment:
```suggestion
                                        />
```
This extra space here doesn't seem necessary. I tested locally with removing it and the spacing seems appropriate without it.

Reply:
removed